### ISSUE TITLE
Semantic-UI Failed to configure for Symfony 4.2.8

### ISSUE THREAD

**Issue by mlopezgva** [phpMv/phpMv-UI#8]
### Semantic-UI Failed to configure for Symfony 4.2.8
YAML syntax error.

### Steps
I put the following in the `config/services.yaml` file:

```yaml
parameters:
    locale: 'en'

    jquery.params:
        semantic: true
        defer: true

services:
    # ... Symfony stuff ...
    # PhpMV-UI (https://phpmv-ui.kobject.net/Index/semantic/31)
    jquery:
        class: Ajax\php\symfony\JsUtils
        arguments: ['%jquery.params%','@router']
        scope: request

    app.default_controller:
        class: AppBundle\Controller\DefaultController
        arguments: ['@service_container','@jquery']

    Ajax\php\symfony\JsUtils: '@jquery'
```

### Expected Result
Loading of the dependencies.

### Actual Result
![image](https://user-images.githubusercontent.com/996070/57467931-c7ce9e80-7283-11e9-9a74-b646a3708be7.png)

### Versions
phpMv 7.2.17
php  7.2.17
php framework Symfony 4.2.8

**Comment by jcheron** [comment]
Hi @mlopezgva ,
It's easier if you use autowiring

I can't give you an example right now, I'm in a meeting, I'll give you this tonight, if you haven't found it before.

**Comment by jcheron** [comment]
you can directly use the class ```Ajax\php\symfony\JquerySemantic``` for this.

**Comment by jcheron** [comment]
Just to illustrate the use of autowiring with phpMv-UI and Symfony:
Create a service inheriting from `JquerySemantic`
```php
namespace App\Services\semantic;

use Ajax\php\symfony\JquerySemantic;

class SemanticGui extends JquerySemantic{
}
```
Check that autowiring is activated in **config/services.yml**:
```yml
services:
    # default configuration for services in *this* file
    _defaults:
        autowire: true      # Automatically injects dependencies in your services.
```
You can then use dependency injection on properties, constructors or setters:
```php
namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use App\Services\semantic\SemanticGui;

BarController extends AbstractController{
	/**
	 * @var SemanticGui
	 */
	protected $gui;

    public function loadViewWithAjaxButtonAction(){
    	$bt=$this->gui->semantic()->htmlButton('button1','a button');
    	$bt->getOnClick("/url",'#responseElement');
    	return $this->gui->renderView("barView.html.twig");
    }
}
```
The `SemanticGui` class can contain methods for creating graphical objects that can be used in controllers:

```php
namespace App\Services\semantic;

use Ajax\php\symfony\JquerySemantic;

class SemanticGui extends JquerySemantic{
	public function buttons(){
		$bts=$this->_semantic->htmlButtonGroups("bts",["Projects","Tags"]);
		$bts->setPropertyValues("data-url", ["projects","tags"]);
		$bts->addIcons(["folder","tags"]);
		return $bts;
	}
}
```
Usage in a controller:
```php
namespace App\Controller;

use App\Services\semantic\SemanticGui;

FooController extends AbstractController{
    public function anAction(SemanticGui $gui){
    	$bts=$gui->buttons();
    	$bts->onClick('console.log("click")');
    	return $this->gui->renderView("fooView.html.twig");
    }
}
```
I usually create a basic Gui service, and derived services to manage the graphic components specific to each model.

**Comment by mlopezgva** [comment]
OK, the loading part seems to be solved. However, now I'm stumbled trying to get some (any) output. Situation:

- services _default autowire has been `on'  all the time.
- Void `App\Services\semantic\SemanticGui` in its place (newly created `src/Services/semantic/` dir)
- rewritten already working controller to:

```php
<?php
namespace App\Controller;

// use Symfony\Component\HttpFoundation\Response;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use App\Services\semantic\SemanticGui;

class CampaignController extends AbstractController
{

	/**
	 * @var SemanticGui
	 */
	protected $gui;

	public function list() {
    	$bt=$this->gui->semantic()->htmlButton('button1','a button');
    	$bt->getOnClick("/url",'#responseElement');
        echo $bt;
    	// return $this->gui->renderView("barView.html.twig");
	}	
}
```

After trying many examples from the PhpMv-Ui doc and the code here, I always get this error:

![image](https://user-images.githubusercontent.com/996070/57525957-6ca4b680-732c-11e9-9f67-0bc805cb3e0f.png)

I may be dumb, or maybe it is because it's Friday, I don't know... O:)

**Comment by jcheron** [comment]
Hi @mlopezgva ,
It seems you forgot a use statement
Maybe ```use Ajax\semantic\htmlcollections\form\HtmlFormInput```
You must use the code completion of your IDE for the uses

**Comment by mlopezgva** [comment]
I tried that. This is a prnScr of the source and the results, together:

![image](https://user-images.githubusercontent.com/996070/57532420-e4c6a880-733b-11e9-8710-f719c7447f7d.png)

I tried with the Http\Response and without; with and without the HtmlFormInput `use` statement; I tried with the use HtmlFormInput as a trait (inside the class)...

EDIT:
I also did a `composer update`. It updated 3 symfony components, but nothing changed.

I'm running out of options. :-(

**Comment by jcheron** [comment]
The code you show me doesn't need to import `HtmlFormInput`
The error is reported in `WelcomeController.php`, line 14

**Comment by mlopezgva** [comment]
Which is the `$bts->onClick()` call.

I also don't understand why is asking for the `FormInput`, since I have not asked for any form... And even if it did ("it's a button, it needs a form" :-D ) it should load it automatically... Right?

To use this library should be a time saver, but I'm loosing a full day with this. :-(

Wow.

I'm not used to symfony (first project ever... maybe the last one, or not... but it didn't begin quite eright :-D )

I cleared the project's cache (`bin/console cache:clear`) and

![image](https://user-images.githubusercontent.com/996070/57534086-8ac7e200-733f-11e9-9d30-1f27c704cd51.png)

Now if shows up.

I'll modify the `Save` action in vim to perform a `cache:clear` every time I save something. :rofl: 

Thank you, and sorry for taking your time. Now I can begin to playaround and see if  get something done yet.

...
EDIT
Well, it did not end too well. I still get this error when executing a `bin/console debug:router`. I'll try to check with symfony...

**Comment by mlopezgva** [comment]
Sorry to bother again...
I've restarted the configuration of the extension. Found that this was commented, because Symfony reports an error:

![image](https://user-images.githubusercontent.com/996070/57539116-10508f80-734a-11e9-90de-e2c48685b950.png)

If I put it like this:

```yaml
parameters:
    locale: 'en'

    jquery.params:
        semantic: true
        defer: true

services:
    # default configuration for services in *this* file
    _defaults:
        autowire: true      # Automatically injects dependencies in your services.
        autoconfigure: true # Automatically registers your services as commands, event subscribers, etc.

    # PhpMV-UI (https://phpmv-ui.kobject.net/Index/semantic/31)
    jquery:
        class: Ajax\php\symfony\JsUtils
        arguments: [%jquery.params%,'@router']
        scope: request
    #
    app.default_controller:
        class: AppBundle\Controller\CampaignController
        arguments: ['@service_container','@jquery']
    Ajax\php\symfony\JsUtils: '@jquery'
```

I get this:

```
In FileLoader.php line 166:
                                                                                                                                                                                                                   
  The file "/home/nicamlg/Projects/CampaignManager/config/services.yaml" does not contain valid YAML: The reserved indicator "%" cannot start a plain scalar; you need to quote the scalar at line 23 (near "argu  
  ments: [%jquery.params%,'@router']") in /home/nicamlg/Projects/CampaignManager/config/services.yaml (which is loaded in resource "/home/nicamlg/Projects/CampaignManager/config/services.yaml").                 
                                                                                                                                                                                                                   

In YamlFileLoader.php line 633:
                                                                                                                                                                                                                   
  The file "/home/nicamlg/Projects/CampaignManager/config/services.yaml" does not contain valid YAML: The reserved indicator "%" cannot start a plain scalar; you need to quote the scalar at line 23 (near "argu  
  ments: [%jquery.params%,'@router']").                                                                                                                                                                            
                                                                                                                                                                                                                   

In Inline.php line 301:
                                                                                                                                                                                                                   
  The reserved indicator "%" cannot start a plain scalar; you need to quote the scalar at line 23 (near "arguments: [%jquery.params%,'@router']"). 
```

![image](https://user-images.githubusercontent.com/996070/57539274-71786300-734a-11e9-9672-d85aaa33f2ad.png)

Which seems to be related in some way to a problem loading some or other class.

**Comment by jcheron** [comment]
with Symfony, it is sometimes necessary to delete cache files manually, outside the symfony console

**Comment by jcheron** [comment]
With autowiring, you can remove configuration lines that generate an error and that were usable with Symfony 3.*

**Comment by jcheron** [comment]
I have a small example of a Symfony project working with phpMv-UI Semantic if you want.
It's just the beginning of a project, but if it helps you:
[Symfony-semantic-sample](https://github.com/s4-dut-info-2018/boards-symfony)

**Comment by mlopezgva** [comment]
Hello again.

I finally discovered the source of the error.
Well, I found it yesterday. In another controller I had that `HtmlFormInput`, but since I was not "loading" that page, I had dismissed it as the problem source. Symfony kind of "compiles" ALL sources (to prepare routes and other stuff), and that was creating the error showed by the interface. Like I said, I am new to Symfony. My mistake.

Now, I have found this add-on is using SemanticUI. Is there any way for it to use [FomanticUI](https://fomantic-ui.com), which has more up-to-date components?

EDIT

I will open a new issue for this. Thank you for your help and patience.
